server/src/util/foodEntryMiddleware.js

```diff
@@ -53,40 +53,176 @@ export const isExpired = (req, res, next) => {
  */
 export const isDistributorId = async (req, res, next) => {
     const { companyId } = req.body;
-    const query = await prisma.distributor.findUnique({
+    const query = await prisma.foodEntry.findUnique({
       where: {
         id: companyId,
       },
     });
     if (query === NULL || query === undefined) {
```

Comment:
change this to lowercase null, I made a mistake in my original middleware. Apply this change to the other instances as well